### PR TITLE
Fixes #2696

### LINKED ISSUE #2696: Third party libraries need to be updated
This is a clean list of all the libraries which need to be updated to their latest version, or at least to the last version which does not break any existing MyBB functionality. Feel free to add others.

- [x] jQuery (1.11.3 –> 1.12.4)
- [x] CodeMirror (unknown –> 5.24.2)
- [x] jQuery Validation (1.14.0 –> 1.16.0)
- [x] jGrowl (1.4.3 –> 1.4.6)
- [x] jQuery modal (0.5.8 –> 0.8.0)
- [x] jQuery Impromptu (6.2.1 –> 6.2.3)
- [x] jQuery cookie (1.4.1 –> JavaScript Cookie 2.1.4)
- [x] Google Recaptcha #2699
- [x] jQuery UI (1.11.2 –> 1.12.1)

Select2 (3.5.3 –> 4.0.3) #2182 (waiting for author's fix of known issue)

### PR DESCRIPTION
**Pull request by Shade-**
#2696 

Libraries updated needing independent testing:

- [x] jQuery
- [x] CodeMirror
- [x] jGrowl
- [x] jQuery modal
- [x] JS Cookies
- [x] Google reCAPTCHA invisible
- [ ] jQuery Validation
- [ ] jQuery UI
- [ ] jQuery Impromptu

### CODE REVIEW

File: jscripts/validate/jquery.validate.min.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
\ No newline at end of file
+ * Copyright (c) 2016 JĆ¶rn Zaefferer; Licensed MIT */
```
Comment (Stefan-ST) [inline jscripts/validate/jquery.validate.min.js]: There seems to be a minor encoding issue.

Comment (Shade-) [inline jscripts/validate/jquery.validate.min.js]: That's an issue with the official library. Good spot.

**Comment by Shade-** [comment]
I have solved the merge conflicts. Still needs proper testing.

**Comment by Cu8eeeR** [comment]
codemirror v5.29 released (would be nice to have latest libraries when be merged - MyBB v1.8.13?)

P.S. Google reCaptcha is working fine for me, good work!

**Comment by Shade-** [comment]
Does the invisible one work? Thank you for your report, I will update CodeMirror ASAP @Cu8eeeR

**Comment by Cu8eeeR** [comment]
Briefly tested and it was fine, anyway it would be great if anybody else from MyBB team can confirm that all is working as expected.

**Comment by skmedix** [comment]
JQuery Validation 1.17.0 and CodeMirror 5.30.0 were released

**Comment by Shade-** [comment]
Solved the conflicts once again and updated CodeMirror and jQuery validation to their latest versions.

**Comment by Eldenroot** [comment]
I have been testing for over 1 month on my testing board, no issues so far, but it needs someone else who can check it. It would be really nice to merge it ASAP to avoid other merge conflicts in future, because it is waste of time...

**Comment by euantorano** [comment]
I will test this weekend, as I would also like to get this merged.

**Comment by Shade-** [comment]
What plugins/libraries have you been testing @Eldenroot? Just to tick the ones that still need some.

**Comment by Eldenroot** [comment]
I applied all of them to my board. For me these should be fine:
- Google reCAPTHA invisible (I need someone to confirm this if it is fine)
- "older" version of CodeMirror
- jQuery
- jGrowl

these should be safe, at least on my board and under my testing circumstances. Only for Google reCAPTHA invisible needs to be confirmed by somebody else, because I had some issues which was solved somehow... I am not sure what was wrong on my installation. :(

**Comment by WildcardSearch** [comment]
I'll try to test today.

**Comment by Shade-** [comment]
Thank you both @Eldenroot and @WildcardSearch. I had a chance to re-test the invisible captcha and it seems I'm not a robot anymore 🎉

**Comment by euantorano** [comment]
@WildcardSearch Have you had a chance to look at this?

**Comment by WildcardSearch** [comment]
So sorry, I looked at this yesterday, but I got busy and forgot to report in.

I can confirm:

jQuery
CodeMirror
jGrowl
jQuery modal
JS Cookies
Google reCAPTCHA invisible

**Comment by Shade-** [comment]
Last 3 remaining:
- jQuery Validation
- jQuery UI
- jQuery Impromptu

**Comment by WildcardSearch** [comment]
I have obligations today, but I will try to look at validation throughout the week when I have time.

**Comment by Eldenroot** [comment]
Guys, did you have time to test them? Would be nice to include this PR into 1.8.13

**Comment by PaulBender** [comment]
I've tested it, and haven't seen anything wrong.

**Comment by euantorano** [comment]
@PaulBender are you happy to merge this then?

**Comment by PaulBender** [comment]
Yes, I'll merge this

**Pull request merged**